src/lib/PostPage.svelte

```diff
@@ -78,8 +124,72 @@
       {title}
     </h1>
   {/if}
+  <div class="meta-data-line">
+    {#if authors.length > 0}
+      <span class="author-information"
+        >Written by {@html prettyNameConcatenation(authors, scrollifyPerson)}</span
+      >
+      |{/if}
+    <span class="publication-time">{formatDate(publish)}</span>
+    <!-- ToDo: Put Post Category Tags here - might make a good component as they are used in three locations including this one -->
+    <!-- <span class="category-labels"></span> -->
+  </div>
   {@html marked(body, { renderer: postRenderer })}
 </article>
+{#if authors.length > 0}
+  <div class="author-box">
+    <div class="author-box-header">
+      {#if authors.length > 2}
+        <h3>About the Authors</h3>
+      {:else}
+        <h3>About the Author</h3>
+      {/if}
+    </div>
+    <div class="author-box-content" bind:this={authorBox}>
+      {#each authors as author}
+        <div class="author-box-person">
+          <img class="author-profile-picture" src={author.avatar} alt={author.name} />
+          <div class="author-box-person-info">
+            <div class="header-line">
+              <h4 class="author-name">
+                {author.name}
+              </h4>
+              <div class="social-links">
+                {#each author.socialLinks ?? [] as social}
+                  <SocialIcon social={social.platform} id={social.id} />
+                {/each}
+              </div>
+            </div>
+            {#if author.bio !== undefined}
+              {@html marked(author.bio)}
+            {:else}
+              <p>Writting for Beast Saber as a guest.</p>
+            {/if}
+
+            <div class="social-links-mobile">
+              {#each author.socialLinks ?? [] as social}
+                <SocialIcon social={social.platform} id={social.id} />
+              {/each}
+            </div>
+          </div>
+        </div>
+      {/each}
+    </div>
+  </div>
+{/if}
+{#if credits.length > 0}
+  {#each credits as singleCredit}
+    <div class="credit-line">
+      Thanks to
```

Comment:
This formatting is very specific IMO, and will get pretty repetitive when there's a few contributors. i.e:

> Thanks to x for proofreading.
> Thanks to y for fact checking.
> Thanks to z for their contribution.

A list would be cleaner in my eyes, fox example:

> **Special thanks:**
> - x: proofreading
> - y: fact checking
> - z

Reply:
Also maybe include this in the card with the authors. At first I thought it was added to the bottom as part of the article.

Reply:
I don't expect there to be many contributors usually. I think the list falls apart when you have at most 1 contributor.

One can also group multiple contributors under the same contribution so it could be:

> Thanks to x, y and z for proofreading and checking.

I'm kinda okay with the repetition if it happens to be multiple. Since they are entire sentences I think it still reads nicer/more appreciative than a list.

I'll look into adding it to the author box.

Reply:
That still leaves the fact that the formatting is very specific. Whoever adds the contributor has to know that they're essentially completing a sentence.

Also I think the list will still look fine with only one contributor. You can even group contributors in there as well:

> **Special thanks:**
> - x, y, and z: proofreading and checking

Reply:
That's true. It's not used anywhere else though nor would I anticipate that to be used anywhere else so I'm fine with it being more specific. 

I'm not sure that your list is more general though. Wouldn't I be able to just put a "for" in there and effectively have the same? I'm trying to find an example that works for the list but not the current implementation, but nothing comes to mind. They might be (close) to equivalent.

There is an explanation in the CMS that details that though:
> The contribution of the contributor(s) to this article. If provided it will be inserted into the articles as "Thanks to {{Contributor}} for {{Contribution}}." If not provided, {{Contribution}} will default to "for their contribution".

Reply:
Perhaps something like:

```
(Special) Thanks to...
    ... x, y, and z for proofreading
```

does the trick for the both of us? It'd take the repetition out of it.

Reply:
> Wouldn't I be able to just put a "for" in there and effectively have the same?
In this example, yes. But there's other ways contributors could be mentioned. For instance:

> **Special thanks:**
> - x: project lead
> - y: lead designer
> - z: emotional support

Those first two wouldn't work when the "for" is there by default, limiting what can be put there.

As for "Special thanks" vs "Thanks to", I don't mind either way. The ellipses look a bit strange to me though, *especially* with only one contributor.

Reply:
That's a good example. Didn't think about role based mentions. I guess they don't make often make sense for articles, but yeah it is more general.

Ellipses would look odd, that's also true. Would you be fine with:
Ellipses and the less general "for" coding as described above for > 1 entries and just "(Special) Thanks to abc for xyz" in the = 1 case or would you insist on lists?

Reply:
Seems to me we're just stuck on that point, so I removed it for now.

This has been the previous code for anyone who wants to implement that later and have a basis for it. Data model remains.

```
{#if credits.length > 0}
  {#each credits as singleCredit}
    <div class="credit-line">
      Thanks to
      {@html prettyNameConcatenation(singleCredit.contributors)}
      {#if singleCredit.contribution}
        for {singleCredit.contribution}.
      {:else}
        for their contribution.
      {/if}
    </div>
  {/each}
{/if}
```